### PR TITLE
CONCF-1008 | Do not save empty sections (without items)

### PR DESCRIPTION
**Pull request by vforge**

### CODE REVIEW

File: extensions/wikia/CuratedContent/CuratedContentHelper.class.php [hunk 1]
```diff
@@ -11,27 +11,32 @@ class CuratedContentHelper {
...
 		if ( !empty( $sections ) && is_array( $sections ) ) {
 			foreach ( $sections as $section ) {
-				$processedSections[ ] = $this->processLogicForSection( $section );
+				$processedSections[] = $this->processLogicForSection( $section );
 			}
 		}
-		return $processedSections;
+
+		// remove null elements from array
+		return array_filter($processedSections, function($var) { return !is_null($var); } );
```
Comment (bognix) [inline extensions/wikia/CuratedContent/CuratedContentHelper.class.php]: white spaces and how about more descriptive name than $var?

**Comment by bognix** [comment]
minor comment, except that LGTM

File: extensions/wikia/CuratedContent/CuratedContentHelper.class.php (hunk 1 above)
Comment (vforge) [inline extensions/wikia/CuratedContent/CuratedContentHelper.class.php]: `$x` ? :P :imp:

**Comment by hakubo** [comment]
yup, some whitespace issues
otherwise LGTM

File: extensions/wikia/CuratedContent/CuratedContentHelper.class.php [hunk 2]
```diff
@@ -11,27 +11,32 @@ class CuratedContentHelper {
...
 				$this->fillItemInfo( $item );
 				$this->processCrop( $item );
 			}
-		}
 
-		return $section;
+			return $section;
+		} else {
+			// return null if we don't have any items inside section
+			return null;
```
Comment (tomasznapieralski) [inline extensions/wikia/CuratedContent/CuratedContentHelper.class.php:32]: What do you think about removing `else` and moving `return` after `if`?

File: extensions/wikia/CuratedContent/CuratedContentHelper.class.php (hunk 1 above)
Comment (bognix) [inline extensions/wikia/CuratedContent/CuratedContentHelper.class.php]: $section?

**Comment by bognix** [comment]
how about adding unit tests to processSection method?

**Comment by vforge** [comment]
@bognix I was thinking what logic would I test, exactly and only thing I could test in `processSection` is proper use of `array_filter`. Do you think it's necessary?

**Comment by bognix** [comment]
Example tests:
- $sections not empty and is an array
- $sections is empty
- $sections not empty and isn't an array
- $sections not empty and is an array
- $processedSections is [null, null]
- $processedSections is [null, 'foo']
- $processedSections is ['foo', 'foo']

**Comment by bognix** [comment]
LGTM

**Comment by tomasznapieralski** [comment]
LGTM

**Pull request merged**